test/geometry_tests.cpp

```diff
@@ -61,105 +94,176 @@ bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe,
 
 }
 
+std::ostream& operator<<(std::ostream& os, const std::vector<Cell>& v){
+	os << "\n";
+	for (const auto& pixel:v) {
+		os << "(" << pixel.x << "," << pixel.y << ")\n";
+	}
+	os << "\n";
+	return os;
+}
+
+inline bool operator==(const Cell &lhs,const Cell &rhs) {
+	return (lhs.x == rhs.x) && (lhs.y == rhs.y);
+}
+
 TEST(Point2DPlanning_Griz_Test, bresenham_conversion_test) {
 	// Get the points for the line connecting (0,0) and (5,0)
-	auto v = bresenham_conversion(0, 0, 5, 0);
+	const auto v = bresenham_conversion(0, 0, 5, 0);
 
 	// Check if the vector size is 6 points 
-	ASSERT_EQ(v.size(), 6);
+	EXPECT_THAT(v.size(), testing::Eq(6));
 
 	// Check if the first point is (0,0)
-	ASSERT_EQ(checkFirstOrLastPoint(0, 0, 0, v), true);
+	EXPECT_THAT(v.front(), testing::ElementsAre(0,0));
```

Comment:
I'm not sure why you're checking this as two end points? Wouldn't you rather
```
const std::vector<Cell> expected{{0, 0}, {1, 0}, {2, 0}, {3, 0}, {4, 0}, {5, 0}};
EXPECT_THAT(v, ::testing::ElementsAreArray(expected));
```

Btw, you can do `using ::testing::ElementsAreArray;` after the headers but before the definitions, so that you can type `ElementsAreArray` directly. In general, don't violate https://abseil.io/tips/153 though I understand the reasons might be a little hard to grok.

Reply:
Changed